executor/src/witgen/jit/interpreter.rs

```diff
@@ -0,0 +1,441 @@
+// TODO: the unused is only here because the interpreter is not integrated in the final code yet
+#![allow(unused)]
+use super::compiler::written_vars_in_effect;
+use super::effect::{Assertion, Effect};
+
+use super::symbolic_expression::{BinaryOperator, BitOperator, SymbolicExpression, UnaryOperator};
+use super::variable::{Cell, Variable};
+use crate::witgen::data_structures::finalizable_data::CompactDataRef;
+use crate::witgen::data_structures::mutable_state::MutableState;
+use crate::witgen::jit::effect::MachineCallArgument;
+use crate::witgen::machines::LookupCell;
+use crate::witgen::QueryCallback;
+use powdr_number::FieldElement;
+
+use std::collections::{BTreeSet, HashMap};
+
+/// Witgen effects compiled into interpreter instructions.
+pub struct EffectsInterpreter<T: FieldElement> {
```

Comment:
InterpreterEffects? InterpreterCode?

I am actually thinking about splitting up `Effects` into `Effects` (used at inference time) and `Code` (used for the final code), since they do not really overlap that much any more. Just as a side note...

Reply:
ah sorry, this is the actual interpreter. all good :)